lib/spack/spack/cmd/install.py

```diff
@@ -289,7 +289,15 @@ def install(parser, args, **kwargs):
                 env.regenerate_views()
             return
         else:
-            tty.die("install requires a package argument or a spack.yaml file")
+            msg = "install requires a package argument or active environment"
+            if 'spack.yaml' in os.listdir(os.getcwd()):
+                # There's a spack.yaml file in the working dir, the user may
+                # have intended to use that
+                msg += "\n\n"
+                msg += "Did you mean to use the environment"
+                msg += " in the current directory?\n"
+                msg += "    Try 'spack -e . install'"
```

Comment:
1. I think this formatting is clearer:

    ```
    Did you mean to install using the `spack.yaml` in this directory?  Try:
        spack env activate .
        spack install
    ```

2. Question: should our standard recommendation be:

    ```
    spack env activate .
    spack install
    ```

    or 

    ```
    spack --env . install
    ```

    I actually think if we're trying to push users towards environments, the first one is better.  The second one, if we like it (though after talking to @becker33 I think it's the less common use case), should at least have `--env` instead of `-e` to make it clearer what the arg does (@adamjstewart's principle of docs having long args).

Reply:
I've changed it to suggest both options, in the order you presented them here.

Reply:
@tgamblin I've changed it to suggest both options, in the order you presented them here. I've also made the wording change.